Fixes and updates

uguraslan opened the pull request:
- As a temporary solution to use `msdf-generator` as a library in other projects, pushed `dist` folder with generated code.
- instead of running msdf-bmfont as an CLI command inside `genFont.ts`, now it uses `generateBMFont` from `msdf-bmfont-xml` package directly. This solves some edge issues related to path resolution of `msdf-bmfont` when there is a complex dependency tree.
- As a temporary solution, instead of using the npm package for `msdf-bmfont-xml`, it uses direct GitHub URL with a fixed commit hash. Because the package maintainers added more binaries for Linux arm64 and MacOS arm64 platforms. But they haven't created a release for it for the last 10 months.
- now the project requires Node 18+ engine.

frank-weindel reviewed (changes requested):
Great work! LGTM in principal. Though the `dist` directory should be removed from the git repo and only published to NPM.

michielvandergeest commented on package.json:
```diff
@@ -15,17 +15,19 @@
   "keywords": [],
   "author": "",
   "license": "ISC",
```
License mismatch with the actual license of the package (Apache 2.0)

uguraslan commented:
I removed the `dist` folder and added `prepublishOnly` script in order to build the project before npm publish. Also updated license info and added license text to all files, if possible.

michielvandergeest reviewed (approved):
Looks good to me now with these last changes!
